planner/core/find_best_task.go

```diff
@@ -1737,8 +1737,8 @@ func (ds *DataSource) crossEstimateRowCount(path *util.AccessPath, conds []expre
 	if len(accessConds) == 0 {
 		return 0, false, corr
 	}
-	ranges, err := ranger.BuildColumnRange(accessConds, ds.ctx, col.RetType, types.UnspecifiedLength)
-	if len(ranges) == 0 || err != nil {
+	ranges, accessConds, _, err := ranger.BuildColumnRange(accessConds, ds.ctx, col.RetType, types.UnspecifiedLength, ds.ctx.GetSessionVars().RangeMaxSize)
```

Comment:
Why ignore the `remainConds` here?

Reply:
The reason is similar to `getMaskAndRanges`. Both functions call `BuildColumnRange` to build ranges so we can estimate row count from ranges. We can ignore the conditions which cannot be used to build ranges. By contrast, when we call `BuildTableRange` in `deriveTablePathStats` to fill the path, we cannot ignore `remainedConds` because we must put `remainedConds` in `Selection` operator.